src/main/scala/com/ing/wbaa/airlock/proxy/api/directive/ProxyDirectives.scala

```diff
@@ -61,5 +62,31 @@ object ProxyDirectives extends LazyLogging {
           }
       }
     }
+
+  /**
+   * Updates the forward headers of a request.
+   */
+  val updateHeadersForRequest: Directive1[HttpRequest] =
+    extractRequest tflatMap { case Tuple1(httpRequest) =>
+      optionalHeaderValueByName("Remote-Address") tflatMap { case Tuple1(remoteAddressHeader) =>
+        optionalHeaderValueByName("X-Forwarded-For") tmap { case Tuple1(xForwardedForHeader) =>
```

Comment:
Remote-Address, X-Forwarded-For, $ffh and others are magic strings. Need to make them constants with understandable names. Could also create a constants class if these definitions are used throughout Airlock